packages/dropdown/src/Dropdown.tsx

```diff
@@ -6,104 +6,107 @@ import { DropdownProps, SelectOptions } from './Dropdown.types';
-    fullWidth && styles.fullWidth,
-  );
+    const classNames = cn(
+      styles.container,
+      disabled && styles.disabled,
+      fullWidth && styles.fullWidth,
+    );
 
-  const arrowClassNames = cn(
-    disabled && styles.caret_disabled,
-    isOpen && styles.caret_up,
-    !isOpen && styles.caret_down,
-  );
+    const arrowClassNames = cn(
+      disabled && styles.caret_disabled,
+      isOpen && styles.caret_up,
+      !isOpen && styles.caret_down,
+    );
 
-  const valueClassNames = cn(
-    disabled && 'disabled',
-    styles.value,
-  );
+    const valueClassNames = cn(disabled && 'disabled', styles.value);
 
-  return (
-    <Component
-      className={classNames}
-      onBlur={onBlurHandler}
-      onClick={onClickHandler}
-      tabIndex={0}
-      {...restProps}
-    >
-      <span className={valueClassNames}>
-        {stateValue?.label}
-      </span>
-      <div className={arrowClassNames}></div>
-      <ul
-        className={`${styles.options} ${isOpen ? styles.show : ''}`}
-        data-test-id={dataTestId}
+    return (
+      <Component
+        className={classNames}
+        onBlur={onBlurHandler}
+        onClick={onClickHandler}
+        ref={ref}
+        tabIndex={0}
+        {...restProps}
       >
-        {options.map(option => (
-          <li
-            className={`${styles.option} ${isOptionSelection(option) ? styles.selected : ''
-              }`}
-            key={option.value}
-            onClick={e => {
-              e.stopPropagation();
-              selectOption(option);
-              setIsOpen(false);
-            }}
-          >
-            <div className={styles.contentContainer}>
-              {option.leftContent && (
-                <span className={styles.leftContent}>{option.leftContent}</span>
-              )}
-              <span className={styles.optionContent}>{option.label}</span>
-              {option.rightContent && (
-                <span className={styles.rightContent}>
-                  {option.rightContent}
-                </span>
-              )}
-            </div>
-          </li>
-        ))}
-      </ul>
-    </Component>
-  );
-}
+        {selectedOption?.label && <span className={valueClassNames}>{selectedOption.label}</span>}
+        <div className={arrowClassNames}></div>
+        <ul
+          className={`${styles.options} ${isOpen ? styles.show : ''}`}
+          data-test-id={dataTestId}
+        >
+          {options.map(option => (
+            <li
+              className={`${styles.option} ${isOptionSelection(option) ? styles.selected : ''
+                }`}
+              key={option.value}
+              onClick={e => {
+                e.stopPropagation();
+                selectOption(option);
+                setIsOpen(false);
+              }}
+            >
+              <div className={styles.contentContainer}>
+                {option.leftContent && (
+                  <span className={styles.leftContent}>
```

Comment:
@hcafaq please check the test coverage

Reply:
converted to draft will push another commit for it